__test__/unit/blockchain_api/scenarios/invalid_repayments.ts

```diff
@@ -0,0 +1,23 @@
+import { RepaymentRouterError } from "src/types";
+import { RepaymentRouterErrorScenario, DEFAULT_REPAYMENT_ARGS } from "./";
+
+export const INVALID_REPAYMENT_SCENARIOS: RepaymentRouterErrorScenario[] = [
+    {
+        description: "the debt agreement does not exist",
+        error: RepaymentRouterError.DEBT_AGREEMENT_NONEXISTENT,
+        ...DEFAULT_REPAYMENT_ARGS,
+        agreementExists: false,
+    },
+    {
+        description: "the payer's balance is insufficient",
+        error: RepaymentRouterError.PAYER_BALANCE_OR_ALLOWANCE_INSUFFICIENT,
+        ...DEFAULT_REPAYMENT_ARGS,
+        isPayerBalanceInsufficient: true,
+    },
+    {
+        description: "the repayment is not in the expected token",
```

Comment:
This error description should be more general -- i.e. `the terms contract has rejected the repayment`